examples/landmark_detection/example_scan.py

```diff
@@ -0,0 +1,218 @@
+        self.transformation = transformation
+        self.scales = scales
+        self.kernel_size = kernel_size
+        self.sigma = sigma
+
+    def run(self, model, dataset):
+        if self.transformation == "scaling":
+            dl = ResizedDataLoader(dataset, scales=self.scales)
+            slicing_fn = f"scaling = {self.scales}"
+        elif self.transformation == "color":
+            dl = ColoredDataLoader(dataset)
+            slicing_fn = f"color = grayscale"
+        elif self.transformation == "blurr":
+            dl = BlurredDataLoader(dataset, self.kernel_size, self.sigma)
+            slicing_fn = f"blurr = kernel {self.kernel_size}, sigma {self.sigma}"
+
+        result = TestDiff(metric=NMEMean, threshold=1).run(
+            model=model,
+            dataloader=dl,
+            dataloader_ref=dataset,
+        )
+        result = result.to_dict()
+        level = IssueLevel.MAJOR if result["metric_value"] > 0 else IssueLevel.MEDIUM
+        issues = []
+
+        issues.append(
+            Issue(
+                model,
+                dataset,
+                level=level,
+                slicing_fn=slicing_fn,
+                group=IssueGroup("Robustness to transformations", "Warning"),
+                meta={
+                    "metric": result["metric"],
+                    "metric_value": result["metric_value"],
+                },
+            )
+        )
+        return issues
+
+
+class HeadPoseDetector:
+    def __init__(self):
+        pass
+
+    def run(self, model, dataset):
+        cached_dl = CachedDataLoader(HeadPoseDataLoader(dataset), cache_size=None, cache_img=False, cache_marks=False)
+
+        head_poses = [("positive roll", self._positive_roll), ("negative roll", self._negative_roll)]
+        issues = []
+
+        for hp in head_poses:
+            dl = FilteredDataLoader(cached_dl, hp[1])
+            result = (
+                TestDiff(metric=NMEMean, threshold=1)
+                .run(
+                    model=model,
+                    dataloader=dl,
+                    dataloader_ref=dl_ref,
+                )
+                .to_dict()
+            )
+
+            level = IssueLevel.MAJOR if result["metric_value"] > 0 else IssueLevel.MEDIUM
+            issues.append(
+                Issue(
+                    model,
+                    dataset,
+                    level=level,
+                    slicing_fn=f"Head Pose = {hp[0]}",
+                    group=IssueGroup("Robustness to Head Pose", "Warning"),
+                    meta={"metric": result["metric"], "metric_value": result["metric_value"], "slice_size": len(dl)},
+                )
+            )
+
+        return issues
+
+    def _positive_roll(self, elt):
+        return elt[2]["headPose"]["roll"] > 0
+
+    def _negative_roll(self, elt):
+        return elt[2]["headPose"]["roll"] < 0
+
+
+class EthnicityDetector:
```

Comment:
should inherit from: https://github.com/Giskard-AI/giskard/blob/1882fb6fa617216266d9a8b1fd7fd6307f43a365/giskard/scanner/registry.py#L11-L14